modules/react/select/lib/hooks/useSelectInput.ts

```diff
@@ -130,14 +138,18 @@ export const useSelectInput = composeHooks(
         },
         textInputProps: {
           ref: textInputRef,
+          value: elemProps.value
+            ? model.navigation.getItem(model.state.selectedIds[0], model).textValue
+            : model.state.selectedIds.length > 0 && model.state.items.length > 0
+            ? model.navigation.getItem(model.state.selectedIds[0], model).textValue
+            : undefined,
```

Comment:
I need someone to verify this logic. When Nicholas worked on this in this [sandbox](https://stackblitz.com/edit/controllable-select-model-6bwrh3?file=src%2FuseControlledSelectModel.ts,src%2FuseSelectInput.ts) it worked. However, it broke the dynamic items story because it wasn't setting the value of the visual input.

Reply:
Also this change now throws the following error because we're updating the value ourselves:
```
Warning: You provided a `value` prop to a form field without an `onChange` handler. This will render a read-only field. If the field should be mutable use `defaultValue`. Otherwise, set either `onChange` or `readOnly`.
```

Reply:
The comment splitting the code gave me trouble trying to understand the logic in its entirety:

```tsx
          value: elemProps.value
            ? model.navigation.getItem(model.state.selectedIds[0], model).textValue
            : model.state.selectedIds.length > 0 && model.state.items.length > 0
            ? model.navigation.getItem(model.state.selectedIds[0], model).textValue
            : undefined,
```

It looks like this code says the following: "if the developer passed a `value`, look up the text value of the id. If not, see if there is selected ids and items, if yes, look up the text value, otherwise send `undefined`".

This doesn't make sense. Are you meaning it to always be a "controlled" input to React? It seems like the following would work better:

```tsx
          model.state.selectedIds.length > 0 && model.state.items.length > 0
            ? model.navigation.getItem(model.state.selectedIds[0], model).textValue
            : undefined,
```

This logic says if there's a selected id and items, the `value` of the visual input should be the text value of the server value. The test for `model.state.items` is to make sure the `getItem` call doesn't fail on an error.